### PR TITLE
Upgrade guide  updates

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**
### Summary
- Added a note about webpacker and zeitwerk.

### CODE REVIEW

File: guides/source/upgrading_ruby_on_rails.md [hunk 1]
```diff
@@ -133,6 +133,41 @@ Action Cable JavaScript API:
...
+```sh
+bundle exec rake zeitwerk:check
+```
+
+Once your application is compatible with Zeitwerk, then you
+can enable it:
+
+```
+# config/application.rb
+config.config.autoloader = :zeitwerk
```
Comment (kaspth) [inline guides/source/upgrading_ruby_on_rails.md]: Double config here.

**Review by kaspth — COMMENTED** [review]
Do we actually need these changes here? This guide should explain the changes required to upgrade, not the new defaults that users can adopt afterwards.

This strikes as suggesting we need a new adopting new defaults guide. That would explain in more detail what the new default is and how an app could use it.

cc @fxn for the Zeitwerk bits.

**Comment by prathamesh-sonpatki** [comment]
I feel we need to mention it in the upgrading guide itself as the guide lists the steps that we need to do to make our app ready for the next Rails version. Perhaps we can add it under a different heading at the end for new defaults?

**Comment by fxn** [comment]
The info about Zeitwerk is not 100% accurate and there is a lot more to say. I'd suggest to delete that part of the patch, since I plan to write that section. Thanks for volunteering anyway @prathamesh-sonpatki.

**Comment by prathamesh-sonpatki** [comment]
Thanks, I have removed it.

File: guides/source/upgrading_ruby_on_rails.md [hunk 2]
```diff
@@ -133,6 +133,22 @@ Action Cable JavaScript API:
   +    ActionCable.logger.enabled = false
   ```
 
+### Using Webpacker
+
+[Webpacker](https://github.com/rails/webpacker)
+is the default JavaScript compiler for Rails 6. But if you
+are upgrading the app, it is not activated by default.
+If you want to use Webpacker, then include it in Gemfile
```
Comment (alexcameron89) [inline guides/source/upgrading_ruby_on_rails.md]: "then include it in **your** Gemfile"

Comment (prathamesh-sonpatki) [reply guides/source/upgrading_ruby_on_rails.md]: Corrected

**Comment by prathamesh-sonpatki** [comment]
Does this look good to merge?

**Review by alexcameron89 — APPROVED** [review]
> Do we actually need these changes here? This guide should explain the changes required to upgrade, not the new defaults that users can adopt afterwards.
> 
> This strikes as suggesting we need a new adopting new defaults guide. That would explain in more detail what the new default is and how an app could use it.
> - @kaspth 

I like the idea of the new defaults either being here (maybe under a **New Defaults** heading) or there being a clear action and link to a new page in the guides that users can find the new defaults. Without a clear guide, I think some people may go to upgrade their applications and not realize there are new features that they can take advantage of.

I think a benefit of having them here in the updates guide is that once I've made it through all of the updates and I've scanned through the new additions that I want to take advantage of, I will be happy knowing I've fully updated to Rails 6.

What are your thoughts @kaspth?

**Comment by prathamesh-sonpatki** [comment]
@kaspth Any thoughts this should be merged? 😄

**Comment by kaspth** [comment]
@prathamesh-sonpatki yes, it should, thanks! Merged it in 39a291ed97

@alexcameron89 I still like the idea of two different guides to hammer down the modern "split" of upgrading a Rails app: first getting to the new version and deploying, then once it's stable considering adopting new defaults.

That way the new defaults guide could also have sections for configs that must be adopted before the next Rails version (although they're rare, but it would be good to force apps for the cookie ones, for instance), versus things that we'll continue to maintain (I think `belongs_to_required_by_default`).

**Pull request closed without merging**